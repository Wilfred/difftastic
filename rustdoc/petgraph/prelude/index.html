<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="Commonly used items."><meta name="keywords" content="rust, rustlang, rust-lang, prelude"><title>petgraph::prelude - Rust</title><link rel="preload" as="font" type="font/woff2" crossorigin href="../../SourceSerif4-Regular.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../FiraSans-Regular.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../FiraSans-Medium.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../SourceCodePro-Regular.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../SourceSerif4-Bold.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../SourceCodePro-Semibold.ttf.woff2"><link rel="stylesheet" href="../../normalize.css"><link rel="stylesheet" href="../../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" href="../../ayu.css" disabled><link rel="stylesheet" href="../../dark.css" disabled><link rel="stylesheet" href="../../light.css" id="themeStyle"><script id="default-settings" ></script><script src="../../storage.js"></script><script defer src="../../main.js"></script><noscript><link rel="stylesheet" href="../../noscript.css"></noscript><link rel="alternate icon" type="image/png" href="../../favicon-16x16.png"><link rel="alternate icon" type="image/png" href="../../favicon-32x32.png"><link rel="icon" type="image/svg+xml" href="../../favicon.svg"></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle">&#9776;</button><a class="sidebar-logo" href="../../petgraph/index.html"><div class="logo-container"><img class="rust-logo" src="../../rust-logo.svg" alt="logo"></div></a><h2 class="location"></h2></nav><nav class="sidebar"><a class="sidebar-logo" href="../../petgraph/index.html"><div class="logo-container"><img class="rust-logo" src="../../rust-logo.svg" alt="logo"></div></a><h2 class="location"><a href="#">Module prelude</a></h2><div class="sidebar-elems"><section><div class="block"><ul><li><a href="#reexports">Re-exports</a></li></ul></div></section></div></nav><main><div class="width-limiter"><div class="sub-container"><a class="sub-logo-container" href="../../petgraph/index.html"><img class="rust-logo" src="../../rust-logo.svg" alt="logo"></a><nav class="sub"><form class="search-form"><div class="search-container"><span></span><input class="search-input" name="search" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"><div id="help-button" title="help" tabindex="-1"><button type="button">?</button></div><div id="settings-menu" tabindex="-1"><a href="../../settings.html" title="settings"><img width="22" height="22" alt="Change settings" src="../../wheel.svg"></a></div></div></form></nav></div><section id="main-content" class="content"><div class="main-heading"><h1 class="fqn"><span class="in-band">Module <a href="../index.html">petgraph</a>::<wbr><a class="mod" href="#">prelude</a><button id="copy-path" onclick="copy_path(this)" title="Copy item path to clipboard"><img src="../../clipboard.svg" width="19" height="18" alt="Copy item path"></button></span></h1><span class="out-of-band"><a class="srclink" href="../../src/petgraph/prelude.rs.html#1-21">source</a> · <a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span></div><details class="rustdoc-toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>Commonly used items.</p>

<div class="example-wrap"><pre class="rust rust-example-rendered"><code><span class="kw">use </span>petgraph::prelude::<span class="kw-2">*</span>;</code></pre></div>
</div></details><h2 id="reexports" class="small-section-header"><a href="#reexports">Re-exports</a></h2><div class="item-table"><div class="item-row"><div class="item-left import-item" id="reexport.DiGraph"><code>pub use crate::graph::<a class="type" href="../graph/type.DiGraph.html" title="type petgraph::graph::DiGraph">DiGraph</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.EdgeIndex"><code>pub use crate::graph::<a class="struct" href="../graph/struct.EdgeIndex.html" title="struct petgraph::graph::EdgeIndex">EdgeIndex</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.Graph"><code>pub use crate::graph::<a class="struct" href="../graph/struct.Graph.html" title="struct petgraph::graph::Graph">Graph</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.NodeIndex"><code>pub use crate::graph::<a class="struct" href="../graph/struct.NodeIndex.html" title="struct petgraph::graph::NodeIndex">NodeIndex</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.UnGraph"><code>pub use crate::graph::<a class="type" href="../graph/type.UnGraph.html" title="type petgraph::graph::UnGraph">UnGraph</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.DiGraphMap"><code>pub use crate::graphmap::<a class="type" href="../graphmap/type.DiGraphMap.html" title="type petgraph::graphmap::DiGraphMap">DiGraphMap</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.GraphMap"><code>pub use crate::graphmap::<a class="struct" href="../graphmap/struct.GraphMap.html" title="struct petgraph::graphmap::GraphMap">GraphMap</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.UnGraphMap"><code>pub use crate::graphmap::<a class="type" href="../graphmap/type.UnGraphMap.html" title="type petgraph::graphmap::UnGraphMap">UnGraphMap</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.StableDiGraph"><code>pub use crate::stable_graph::<a class="type" href="../stable_graph/type.StableDiGraph.html" title="type petgraph::stable_graph::StableDiGraph">StableDiGraph</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.StableGraph"><code>pub use crate::stable_graph::<a class="struct" href="../stable_graph/struct.StableGraph.html" title="struct petgraph::stable_graph::StableGraph">StableGraph</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.StableUnGraph"><code>pub use crate::stable_graph::<a class="type" href="../stable_graph/type.StableUnGraph.html" title="type petgraph::stable_graph::StableUnGraph">StableUnGraph</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.Bfs"><code>pub use crate::visit::<a class="struct" href="../visit/struct.Bfs.html" title="struct petgraph::visit::Bfs">Bfs</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.Dfs"><code>pub use crate::visit::<a class="struct" href="../visit/struct.Dfs.html" title="struct petgraph::visit::Dfs">Dfs</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.DfsPostOrder"><code>pub use crate::visit::<a class="struct" href="../visit/struct.DfsPostOrder.html" title="struct petgraph::visit::DfsPostOrder">DfsPostOrder</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.Directed"><code>pub use crate::<a class="enum" href="../enum.Directed.html" title="enum petgraph::Directed">Directed</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.Direction"><code>pub use crate::<a class="enum" href="../enum.Direction.html" title="enum petgraph::Direction">Direction</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.Incoming"><code>pub use crate::<a class="enum" href="../enum.Direction.html" title="enum petgraph::Direction">Incoming</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.Outgoing"><code>pub use crate::<a class="enum" href="../enum.Direction.html" title="enum petgraph::Direction">Outgoing</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.Undirected"><code>pub use crate::<a class="enum" href="../enum.Undirected.html" title="enum petgraph::Undirected">Undirected</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.EdgeRef"><code>pub use crate::visit::<a class="trait" href="../visit/trait.EdgeRef.html" title="trait petgraph::visit::EdgeRef">EdgeRef</a>;</code></div></div></div></section></div></main><div id="rustdoc-vars" data-root-path="../../" data-current-crate="petgraph" data-themes="ayu,dark,light" data-resource-suffix="" data-rustdoc-version="1.65.0 (897e37553 2022-11-02)" ></div></body></html>