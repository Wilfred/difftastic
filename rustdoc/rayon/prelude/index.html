<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="The rayon prelude imports the various `ParallelIterator` traits. The intention is that one can include `use rayon::prelude::*` and have easy access to the various traits and methods you will need."><meta name="keywords" content="rust, rustlang, rust-lang, prelude"><title>rayon::prelude - Rust</title><link rel="preload" as="font" type="font/woff2" crossorigin href="../../SourceSerif4-Regular.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../FiraSans-Regular.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../FiraSans-Medium.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../SourceCodePro-Regular.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../SourceSerif4-Bold.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../SourceCodePro-Semibold.ttf.woff2"><link rel="stylesheet" href="../../normalize.css"><link rel="stylesheet" href="../../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" href="../../ayu.css" disabled><link rel="stylesheet" href="../../dark.css" disabled><link rel="stylesheet" href="../../light.css" id="themeStyle"><script id="default-settings" ></script><script src="../../storage.js"></script><script defer src="../../main.js"></script><noscript><link rel="stylesheet" href="../../noscript.css"></noscript><link rel="alternate icon" type="image/png" href="../../favicon-16x16.png"><link rel="alternate icon" type="image/png" href="../../favicon-32x32.png"><link rel="icon" type="image/svg+xml" href="../../favicon.svg"></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle">&#9776;</button><a class="sidebar-logo" href="../../rayon/index.html"><div class="logo-container"><img class="rust-logo" src="../../rust-logo.svg" alt="logo"></div></a><h2 class="location"></h2></nav><nav class="sidebar"><a class="sidebar-logo" href="../../rayon/index.html"><div class="logo-container"><img class="rust-logo" src="../../rust-logo.svg" alt="logo"></div></a><h2 class="location"><a href="#">Module prelude</a></h2><div class="sidebar-elems"><section><div class="block"><ul><li><a href="#reexports">Re-exports</a></li><li><a href="#traits">Traits</a></li></ul></div></section></div></nav><main><div class="width-limiter"><div class="sub-container"><a class="sub-logo-container" href="../../rayon/index.html"><img class="rust-logo" src="../../rust-logo.svg" alt="logo"></a><nav class="sub"><form class="search-form"><div class="search-container"><span></span><input class="search-input" name="search" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"><div id="help-button" title="help" tabindex="-1"><button type="button">?</button></div><div id="settings-menu" tabindex="-1"><a href="../../settings.html" title="settings"><img width="22" height="22" alt="Change settings" src="../../wheel.svg"></a></div></div></form></nav></div><section id="main-content" class="content"><div class="main-heading"><h1 class="fqn"><span class="in-band">Module <a href="../index.html">rayon</a>::<wbr><a class="mod" href="#">prelude</a><button id="copy-path" onclick="copy_path(this)" title="Copy item path to clipboard"><img src="../../clipboard.svg" width="19" height="18" alt="Copy item path"></button></span></h1><span class="out-of-band"><a class="srclink" href="../../src/rayon/prelude.rs.html#1-17">source</a> · <a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span></div><details class="rustdoc-toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>The rayon prelude imports the various <code>ParallelIterator</code> traits.
The intention is that one can include <code>use rayon::prelude::*</code> and
have easy access to the various traits and methods you will need.</p>
</div></details><h2 id="reexports" class="small-section-header"><a href="#reexports">Re-exports</a></h2><div class="item-table"><div class="item-row"><div class="item-left import-item" id="reexport.FromParallelIterator"><code>pub use crate::iter::<a class="trait" href="../iter/trait.FromParallelIterator.html" title="trait rayon::iter::FromParallelIterator">FromParallelIterator</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.IndexedParallelIterator"><code>pub use crate::iter::<a class="trait" href="../iter/trait.IndexedParallelIterator.html" title="trait rayon::iter::IndexedParallelIterator">IndexedParallelIterator</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.IntoParallelIterator"><code>pub use crate::iter::<a class="trait" href="../iter/trait.IntoParallelIterator.html" title="trait rayon::iter::IntoParallelIterator">IntoParallelIterator</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.IntoParallelRefIterator"><code>pub use crate::iter::<a class="trait" href="../iter/trait.IntoParallelRefIterator.html" title="trait rayon::iter::IntoParallelRefIterator">IntoParallelRefIterator</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.IntoParallelRefMutIterator"><code>pub use crate::iter::<a class="trait" href="../iter/trait.IntoParallelRefMutIterator.html" title="trait rayon::iter::IntoParallelRefMutIterator">IntoParallelRefMutIterator</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.ParallelDrainFull"><code>pub use crate::iter::<a class="trait" href="../iter/trait.ParallelDrainFull.html" title="trait rayon::iter::ParallelDrainFull">ParallelDrainFull</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.ParallelDrainRange"><code>pub use crate::iter::<a class="trait" href="../iter/trait.ParallelDrainRange.html" title="trait rayon::iter::ParallelDrainRange">ParallelDrainRange</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.ParallelExtend"><code>pub use crate::iter::<a class="trait" href="../iter/trait.ParallelExtend.html" title="trait rayon::iter::ParallelExtend">ParallelExtend</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.ParallelIterator"><code>pub use crate::iter::<a class="trait" href="../iter/trait.ParallelIterator.html" title="trait rayon::iter::ParallelIterator">ParallelIterator</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.ParallelSlice"><code>pub use crate::slice::<a class="trait" href="../slice/trait.ParallelSlice.html" title="trait rayon::slice::ParallelSlice">ParallelSlice</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.ParallelSliceMut"><code>pub use crate::slice::<a class="trait" href="../slice/trait.ParallelSliceMut.html" title="trait rayon::slice::ParallelSliceMut">ParallelSliceMut</a>;</code></div></div><div class="item-row"><div class="item-left import-item" id="reexport.ParallelString"><code>pub use crate::str::<a class="trait" href="../str/trait.ParallelString.html" title="trait rayon::str::ParallelString">ParallelString</a>;</code></div></div></div><h2 id="traits" class="small-section-header"><a href="#traits">Traits</a></h2><div class="item-table"><div class="item-row"><div class="item-left module-item"><a class="trait" href="trait.ParallelBridge.html" title="rayon::prelude::ParallelBridge trait">ParallelBridge</a></div><div class="item-right docblock-short">Conversion trait to convert an <code>Iterator</code> to a <code>ParallelIterator</code>.</div></div></div></section></div></main><div id="rustdoc-vars" data-root-path="../../" data-current-crate="rayon" data-themes="ayu,dark,light" data-resource-suffix="" data-rustdoc-version="1.65.0 (897e37553 2022-11-02)" ></div></body></html>