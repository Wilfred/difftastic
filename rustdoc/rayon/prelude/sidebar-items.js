window.SIDEBAR_ITEMS = {"trait":[["ParallelBridge","Conversion trait to convert an `Iterator` to a `ParallelIterator`."]]};