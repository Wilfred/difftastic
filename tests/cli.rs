use std::process::Command;

use assert_cmd::prelude::*;
use predicates::prelude::*;

fn find_runner() -> Option<String> {
    for (key, value) in std::env::vars() {
        if key.starts_with("CARGO_TARGET_") && key.ends_with("_RUNNER") && !value.is_empty() {
            return Some(value);
        }
    }
    None
}

// Sample code from
// https://github.com/assert-rs/assert_cmd/issues/139, supports
// cross-compiled binaries.
fn get_base_command() -> Command {
    let mut cmd;
    let path = assert_cmd::cargo::cargo_bin("difft");
    if let Some(runner) = find_runner() {
        let mut runner = runner.split_whitespace();
        cmd = Command::new(runner.next().unwrap());
        for arg in runner {
            cmd.arg(arg);
        }
        cmd.arg(path);
    } else {
        cmd = Command::new(path);
    }
    cmd
}

#[test]
fn no_such_files() {
    let mut cmd = get_base_command();

    cmd.arg("no_such_file").arg("no_such_file_either");
    cmd.assert().failure().code(2);
}

#[test]
fn inline() {
    let mut cmd = get_base_command();

    cmd.arg("--display=inline")
        .arg("sample_files/simple_1.js")
        .arg("sample_files/simple_2.js");
    cmd.assert().success();
}

#[test]
fn binary_changed() {
    let mut cmd = get_base_command();

    cmd.arg("--display=inline")
        .arg("img/logo.png")
        .arg("/dev/null");

    let predicate_fn = predicate::str::contains("Binary contents changed");
    cmd.assert().stdout(predicate_fn);
}

#[test]
fn has_changes_default_exit_code() {
    let mut cmd = get_base_command();

    cmd.arg("sample_files/simple_1.js")
        .arg("sample_files/simple_2.js");
    cmd.assert().success();
}

#[test]
fn has_changes_requested_exit_code() {
    let mut cmd = get_base_command();

    cmd.arg("--exit-code")
        .arg("sample_files/simple_1.js")
        .arg("sample_files/simple_2.js");
    cmd.assert().failure().code(1);
}

#[test]
fn ignore_comments() {
    let mut cmd = get_base_command();

    cmd.arg("--ignore-comments")
        .arg("sample_files/comma_and_comment_1.js")
        .arg("sample_files/comma_and_comment_2.js");

    let predicate_fn = predicate::str::contains("No syntactic changes");
    cmd.assert().stdout(predicate_fn);
}

#[test]
fn check_only() {
    let mut cmd = get_base_command();

    cmd.arg("--check-only")
        .arg("sample_files/simple_1.js")
        .arg("sample_files/simple_2.js");

    let predicate_fn = predicate::str::contains("Has syntactic changes");
    cmd.assert().stdout(predicate_fn);
}

#[test]
fn check_only_text_file() {
    let mut cmd = get_base_command();

    cmd.arg("--check-only")
        .arg("sample_files/text_1.txt")
        .arg("sample_files/text_2.txt");

    let predicate_fn = predicate::str::contains("Has changes");
    cmd.assert().stdout(predicate_fn);
}

#[test]
fn makefile_text_as_atom() {
    let mut cmd = get_base_command();

    cmd.arg("sample_files/cli_tests/makefile_1.mk")
        .arg("sample_files/cli_tests/makefile_2.mk");

    let predicate_fn = predicate::str::contains("CCFLAGS");
    cmd.assert().stdout(predicate_fn);
}

#[test]
fn yaml_parse_errors() {
    let mut cmd = get_base_command();

    cmd.arg("sample_files/cli_tests/bad_yaml_1.yml")
        .arg("sample_files/cli_tests/bad_yaml_2.yml");

    let predicate_fn = predicate::str::contains("exceeded DFT_PARSE_ERROR_LIMIT");
    cmd.assert().stdout(predicate_fn);
}

#[test]
fn list_languages() {
    let mut cmd = get_base_command();

    cmd.arg("--list-languages");

    let predicate_fn = predicate::str::contains("TOML");
    cmd.assert().stdout(predicate_fn);

    let predicate_fn = predicate::str::contains("*.toml");
    cmd.assert().stdout(predicate_fn);
}

#[test]
fn test_mime_type_false_positive() {
    let mut cmd = get_base_command();

    cmd.arg("sample_files/cli_tests/x_mod_false_positive.js")
        .arg("sample_files/cli_tests/empty.js");

    let predicate_fn = predicate::str::contains("JavaScript");
    cmd.assert().stdout(predicate_fn);
}

#[test]
fn slightly_invalid_utf8() {
    let mut cmd = get_base_command();

    cmd.arg("sample_files/cli_tests/MainWindowViewModel.cs")
        .arg("/dev/null");

    let predicate_fn = predicate::str::contains("using");
    cmd.assert().stdout(predicate_fn);
}

#[test]
fn directory_arguments() {
    let mut cmd = get_base_command();

    cmd.arg("sample_files/dir_1").arg("sample_files/dir_2");

    let predicate_fn =
        predicate::str::contains("only_in_1.c").and(predicate::str::contains("only_in_2.rs"));
    cmd.assert().stdout(predicate_fn);
}

#[test]
fn git_style_arguments_rename() {
    let mut cmd = get_base_command();

    cmd.arg("elisp_oldname.el")
        .arg("sample_files/elisp_1.el")
        .arg("lhs_hash_placeholder")
        .arg("lhs_mode_placeholder")
        .arg("sample_files/elisp_2.el")
        .arg("rhs_hash_placeholder")
        .arg("rhs_mode_placeholder")
        .arg("elisp_newname.el")
        .arg("similarity_placeholder");
    let predicate_fn = predicate::str::contains("Renamed");
    cmd.assert().stdout(predicate_fn);
}

#[test]
fn git_style_arguments_new_file() {
    let mut cmd = get_base_command();

    cmd.arg("simple.txt")
        .arg("/dev/null")
        .arg(".")
        .arg(".")
        .arg("sample_files/simple_1.txt")
        .arg("abcdef1234")
        .arg("100644");
    let predicate_fn = predicate::str::contains("File permissions changed").not();
    cmd.assert().stdout(predicate_fn);
}

#[test]
fn drop_different_path_starts() {
    let mut cmd = get_base_command();

    cmd.arg("sample_files/dir_1/clojure.clj")
        .arg("sample_files/dir_2/clojure.clj");
    let predicate_fn = predicate::str::contains("dir_2/clojure.clj").not();
    cmd.assert().stdout(predicate_fn);
}

#[test]
fn dump_tree_sitter() {
    let mut cmd = get_base_command();

    cmd.arg("--dump-ts")
        .arg("sample_files/simple_1.js")
        .arg("sample_files/simple_2.js");
    cmd.assert().success();
}

#[test]
fn dump_syntax() {
    let mut cmd = get_base_command();

    cmd.arg("--dump-syntax")
        .arg("sample_files/simple_1.js")
        .arg("sample_files/simple_2.js");
    cmd.assert().success();
}

#[test]
fn walk_hidden_items() {
    let mut cmd = get_base_command();

    cmd.args(["sample_files/hidden_1", "sample_files/hidden_2"]);

    let predicate_fn =
        predicate::str::contains(format!(".hidden{}doc.txt", std::path::MAIN_SEPARATOR))
            .and(predicate::str::contains(".hidden.txt"))
            .and(predicate::str::contains("before"))
            .and(predicate::str::contains("after"));
    cmd.assert().stdout(predicate_fn);
}
