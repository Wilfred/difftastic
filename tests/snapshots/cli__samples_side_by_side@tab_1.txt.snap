---
source: tests/cli.rs
expression: stdout
input_file: sample_files/tab_1.txt
---
[1m[93msample_files/tab_2.txt[39m[0m[2m --- Text[0m
[91;1m1 [0m[2m  [0m[91;1mexport[0m[91;1m [0m[91;1mconst[0m[91;1m [0m[91;1mORIGIN[0m[91;1m [0m[91;1m=[0m[91;1m [0m[91;1mnew[0m[91;1m [0m[91;1mURL[0m[91;1m([0m
[91;1m2 [0m[2m  [0m[91;1m    [0m[91;1menv[0m[91;1m.[0m[91;1mVERCEL_ENV[0m[91;1m [0m[91;1m=[0m[91;1m=[0m[91;1m=[0m[91;1m [0m[91;1m"[0m[91;1mproduction[0m[91;1m"[0m
[91;1m3 [0m[2m  [0m[91;1m    [0m[91;1m    [0m[91;1m?[0m[91;1m [0m[91;1m"[0m[91;1mhttps[0m[91;1m:[0m[91;1m/[0m[91;1m/[0m[91;1malpha[0m[91;1m.[0m[91;1msweets[0m[91;1m.[0m[91;1mcommunity[0m[91;1m"[0m
[91;1m4 [0m[2m  [0m[91;1m    [0m[91;1m    [0m[91;1m:[0m[91;1m [0m[91;1menv[0m[91;1m.[0m[91;1mVERCEL_ENV[0m[91;1m [0m[91;1m=[0m[91;1m=[0m[91;1m=[0m[91;1m [0m[91;1m"[0m[91;1mpreview[0m[91;1m"[0m
[91;1m5 [0m[2m  [0m[91;1m    [0m[91;1m    [0m[91;1m    [0m[91;1m?[0m[91;1m [0m[91;1m`[0m[91;1mhttps[0m[91;1m:[0m[91;1m/[0m[91;1m/[0m[91;1m$[0m[91;1m{[0m[91;1menv[0m[91;1m.[0m[91;1mVERCEL_URL[0m[91;1m}[0m[91;1m`[0m
[91;1m6 [0m[2m  [0m[91;1m    [0m[91;1m    [0m[91;1m    [0m[91;1m:[0m[91;1m [0m[91;1m"[0m[91;1mhttp[0m[91;1m:[0m[91;1m/[0m[91;1m/[0m[91;1mlocalhost[0m[91;1m:[0m[91;1m3000[0m[91;1m"[0m
[91;1m7 [0m[2m  [0m[91;1m)[0m[91;1m;[0m
[91;1m8 [0m[2m  [0m
[2m9 [0m[2m1 [0mexport default env;
